File: pinyougou-manager-web/src/main/webapp/js/service/itemCatService.js
app.service("itemCatService",function ($http) {

    this.findByParentId = function (parentId) {
        return $http.get("../itemCat/findByParentId.do?parentId="+parentId);
    }

    //新增
    this.save = function (itemCat) {
        return $http.post("../itemCat/add.do",itemCat);
    }
    //修改
    this.update = function (itemCat) {
       return $http.post("../itemCat/update.do",itemCat);
    }

    this.findItemCatById = function (id) {
        return $http.get("../itemCat/findItemCat.do?id=" + id);
    }
})

File: pinyougou-sellergoods-interface/src/main/java/com/pinyougou/sellergoods/service/ItemCatService.java
package com.pinyougou.sellergoods.service;

import com.pinyougou.pojo.TbItemCat;
import entity.PygResult;

import java.util.List;
import java.util.Map;

/**
 * itemCat分类表
 */
public interface ItemCatService {

    public List<TbItemCat> findByParentId(Long parentId);

    //新增分类
    public void insert(TbItemCat tbItemCat);

    public void update(TbItemCat tbItemCat);

    public TbItemCat findById(long id);

}
